### ISSUE TITLE
Allow setting the global state key

### ISSUE THREAD

**Issue by kuon** [gcazaciuc/redux-fractal#10]
I have the following use case:

```
local({
  key: (props, context) => "myInput"
})(
local({
  key: (props, context) => "myInput",
  namespace: "ui"
}))(MyComponent)
```

The second call should store the component state into `state.ui.myInput`. While the first one uses `state.local.myInput` as present.

The idea is that I might want to serialize `state.ui` into `localStorage` for example, while the other state gets another treatment.

Having another namespace let me handle `state.ui` in a single global redux middleware.

**Comment by gcazaciuc** [comment]
Sounds fishy that you are double wrapping the component in local unless you want to pull data from different stores(in which case you could checkout if maybe a 'connect' isn't better as it allows you to read from state.local.myInput dirrectly and from any other store).

Also i believe there might be some confusion around the behaviour for comps not having a 'createStore' method:
In your example these connect directly to a store named myInput(both of them). The component that defines the store is the owner of it controlling where the state gets stored, how long it should persist etc.

I'm not seeing a good/clear use case to add another config option and I don't imagine how it would work.

**Comment by kuon** [comment]
My example is not clear, it should contain `createStore`.

**Comment by gcazaciuc** [comment]
We could use the key as a namespace, eg allow keys such as 'namespace.mycomp'(eg dot separatted) without introducing new concepts to the API, which is what i generally prefer. Also i don't believe it's a very common use case what you've mentioned but i could be mistaken.

**Comment by kuon** [comment]
Yeah, supporting dot in the key would solve it without introducing new API.

**Comment by gcazaciuc** [comment]
I don't think I'll have time  to work on this feature specifically right now, if you would like and have the time  you can put up a PR and I'll merge it.
